src/main/java/org/dataconservancy/packaging/tool/cli/RdfContentProvider.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright 2016 Johns Hopkins University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dataconservancy.packaging.tool.cli;
+
+import org.apache.jena.rdf.model.*;
+
+import org.dataconservancy.packaging.shared.ContentProvider;
+import org.dataconservancy.packaging.tool.model.ipm.FileInfo;
+import org.dataconservancy.packaging.tool.model.ipm.Node;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+/**
+ * Implementation of ContentProvider for RDF sourced data.
+ * Created by Ben Trumbore on 12/1/2016.
+ */
+public class RdfContentProvider extends ContentProvider {
+
+    private Logger  LOG = LoggerFactory.getLogger(RdfContentProvider.class);
+    private Model   domainObjects = null;
+    private URI     contentPath;
+
+    /**
+     * Create a content provider that can be passed to IpmPackager.buildPackage.
+     * @param domainObjs The RDF model of the domain objects.
+     * @param contentURI A URL to the files corresponding to the domain model.
+     */
+    public RdfContentProvider(Model domainObjs, URI contentURI) {
+        domainObjects = domainObjs;
+        contentPath = contentURI;
+    }
+
+    /**
+     * Returns the domain model provided in the constructor.
+     * @return The Model representing the domain objects
+     */
+    public Model getDomainModel() {
+        return domainObjects;
+    }
+
+    // Define properties that will be inspected for some RDF nodes.
+    private static Property hasType     = ResourceFactory.createProperty("http://www.w3.org/1999/02/22-rdf-syntax-ns#type");
```

Comment:
I wonder if these `Property` objects deserve to be constants somewhere.  I would think at least their URI's do.